### PR TITLE
ddl: refine error message for expression index

### LINKED ISSUE #26797: JSON expression error need to refine
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
``` SQL
mysql> create table t(a int, index ((cast(a as JSON))));
ERROR 3152 (42000): JSON column '_V$_expression_index_0' cannot be used in key specification.
```

### 2. What did you expect to see? (Required)
ERROR 3753 (42000): Cannot create a functional index on a function that returns a JSON or GEOMETRY value.
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #26797 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
 'None' 
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]

**Comment by djshow832** [comment]
[2021-08-02T09:49:13.850Z]   ✘  https://revive.run/r#exported  exported var ErrFunctionalIndexOnJSONOrGeometryFunction should have comment or be unexported  

[2021-08-02T09:49:13.850Z]   ddl/error.go:293:2

File: ddl/error.go [hunk 1]
```diff
@@ -289,5 +289,6 @@ var (
 	errUnsupportedClusteredSecondaryKey = dbterror.ClassDDL.NewStdErr(mysql.ErrUnsupportedDDLOperation, parser_mysql.Message("CLUSTERED/NONCLUSTERED keyword is only supported for primary key", nil))
 
 	// ErrInvalidAttributesSpec is returned when meeting invalid attributes.
-	ErrInvalidAttributesSpec = dbterror.ClassDDL.NewStd(mysql.ErrInvalidAttributesSpec)
+	ErrInvalidAttributesSpec                   = dbterror.ClassDDL.NewStd(mysql.ErrInvalidAttributesSpec)
+	ErrFunctionalIndexOnJSONOrGeometryFunction = dbterror.ClassDDL.NewStd(mysql.ErrFunctionalIndexOnJSONOrGeometryFunction)
```
Comment (tangenta) [inline ddl/error.go:293]: Exported fields need comment.

**Pull request merged**